object_tracker/kalman_filter.py

```diff
@@ -0,0 +1,88 @@
+        x_k = x_det
+        '''
+
+        self._params = params
+        kalman = cv2.KalmanFilter(6, 4)
+        kalman.statePost = state.astype(np.float32)
+        kalman.measurementMatrix = np.array([[1, 0, 0, 0, 0, 0],
+                                             [0, 1, 0, 0, 0, 0],
+                                             [0, 0, 0, 0, 1, 0],
+                                             [0, 0, 0, 0, 0, 1]], np.float32)
+        d = self._params.drag_factor
+        kalman.transitionMatrix = np.array([[1, 0, 1, 0, 0, 0],
+                                            [0, 1, 0, 1, 0, 0],
+                                            [0, 0, d, 0, 0, 0],
+                                            [0, 0, 0, d, 0, 0],
+                                            [0, 0, 0, 0, 1, 0],
+                                            [0, 0, 0, 0, 0, 1]], np.float32)
+
+        kalman.controlMatrix = None
+
+        kalman.processNoiseCov = np.array([[self._params.motion_noise[0], 0, 0, 0, 0, 0],
+                                           [0, self._params.motion_noise[0], 0, 0, 0, 0],
+                                           [0, 0, self._params.motion_noise[1], 0, 0, 0],
+                                           [0, 0, 0, self._params.motion_noise[1], 0, 0],
+                                           [0,0,0,0,self._params.box_noise,0],
+                                           [0,0,0,0,0,self._params.box_noise]], np.float32)
+        kalman.measurementNoiseCov = np.array([[self._params.measurement_noise, 0, 0, 0],
+                                               [0, self._params.measurement_noise, 0, 0],
+                                               [0,0,self._params.box_noise,0],
+                                               [0,0,0,self._params.box_noise]], np.float32)
+        kalman.errorCovPre = np.array([[self._params.initial_estimate_error[0], 0, 0, 0, 0, 0],
+                                           [0, self._params.initial_estimate_error[0], 0, 0, 0, 0],
+                                           [0, 0, self._params.initial_estimate_error[1], 0, 0, 0],
+                                           [0, 0, 0, self._params.initial_estimate_error[1], 0, 0],
+                                           [0, 0, 0, 0, self._params.initial_estimate_error[2], 0],
+                                           [0, 0, 0, 0, 0, self._params.initial_estimate_error[2]]], np.float32)
+        self._kf = kalman
+
+    def correct(self, measurement):
+        self._kf.measurementNoiseCov = np.array([[self._params.measurement_noise, 0, 0, 0],
+                                               [0, self._params.measurement_noise, 0, 0],
+                                               [0,0,self._params.box_noise,0],
+                                               [0,0,0,self._params.box_noise]], np.float32)
+        return self._kf.correct(measurement.astype(np.float32))[[0,1,4,5]].astype(float).flatten()
+
+    def correct1(self, measurement):
```

Comment:
What is the difference between correct and correct1? Do we need both?

Reply:
kalman is irrelevant.. it is from MOSH